tests/test_autonomy/test_cli/test_analyse/test_check_dialogues.py

```diff
@@ -0,0 +1,188 @@
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=())
+
+        assert result.exit_code == 0, result.stdout
+
+    def test_analyse_fail_dialogues_does_not_exist(self) -> None:
+        """Test analyse."""
+
+        self.make_dummy_skill(
+            config_data=[],
+            dialogues=[],
+        )
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=())
+
+        assert result.exit_code == 1, result.stdout
+        assert "dialogue file does not exist" in result.output
+
+    def test_analyse_fail_wrong_class_name(self) -> None:
+        """Test analyse."""
+
+        config_data = [
+            ("abci_dialogues", "AbciDialogue"),
+        ]
+
+        self.make_dummy_skill(
+            config_data=config_data,
+            dialogues=["AbciDialogue"],
+        )
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=())
+
+        assert result.exit_code == 1, result.stdout
+        assert (
+            "Class name for a dialogue should end with `Dialogues`; dialogue=abci_dialogues; class=AbciDialogue"
+            in result.output
+        )
+
+    def test_analyse_fail_wrong_dialogue_name(self) -> None:
+        """Test analyse."""
+
+        config_data = [
+            ("abci_dialogue", "AbciDialogues"),
+        ]
+
+        self.make_dummy_skill(
+            config_data=config_data,
+            dialogues=["AbciDialogues"],
+        )
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=())
+
+        assert result.exit_code == 1, result.stdout
+        assert (
+            "Dialogue name should end with `dialogues`; dialogue=abci_dialogue; class=AbciDialogues"
+            in result.output
+        )
+
+    def test_analyse_fail_common_dialogue_not_found(self) -> None:
+        """Test analyse."""
+
+        config_data = [
+            ("abci_dialogues", "AbciDialogues"),
+        ]
+
+        self.make_dummy_skill(
+            config_data=config_data,
+            dialogues=["AbciDialogues"],
+        )
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=("-d", "some_dialogues"))
+
+        assert result.exit_code == 1, result.stdout
+        assert "Common dialogue 'some_dialogues' is not defined" in result.output
+
+    def test_analyse_fail_common_dialogue_class_not_found(self) -> None:
+        """Test analyse."""
+
+        config_data = [
+            ("abci_dialogues", "AbciDialogues"),
+            ("some_dialogues", "SomeDialogues"),
+        ]
+
+        self.make_dummy_skill(
+            config_data=config_data,
+            dialogues=["AbciDialogues"],
+        )
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=("-d", "some_dialogues"))
+
+        assert result.exit_code == 1, result.stdout
+        assert "dialogue SomeDialogues declared in" in result.output
+        assert "is missing from" in result.output
```

Comment:
Lots of code dup. These could be a single parametrized test.

Reply:
I did not parametrise the tests because it makes it hard to debug when there's a failure for on particular edge case, instead I extracted the common logic and automated the test [6f79ed0](https://github.com/valory-xyz/open-autonomy/pull/1786/commits/6f79ed093cca649e377e2c41c3da9445ff116f63)